### PR TITLE
interfaces/builtin/block_devices: allow blkid to print block device attributes

### PR DESCRIPTION
**Pull request by woodrow-shen**
To allow blkid to visit block_devices attributes via low level probing, so that
we can get information like partition table type or PART_ENTRY_UUID
which could be useful for script based snap without parsing
/run/udev/data/b{major}:{minor}.

Signed-off-by: Hsieh-Tseng Shen <woodrow.shen@canonical.com>

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Looks good, thank you

File: interfaces/builtin/block_devices.go
Comment (mardy) [inline interfaces/builtin/block_devices.go:104]: I don't think that the `r` permission is needed; can you please check if it works without it?

**Review by bboozzoo — APPROVED** [review]
Thanks

File: interfaces/builtin/block_devices.go
Comment (mvo5) [reply interfaces/builtin/block_devices.go:104]: It seems to be the default in e.g. `interfaces/apparmor/template.go` so at least there seems to be no harm in adding it.

Comment (anonymouse64) [reply interfaces/builtin/block_devices.go:104]: We have some cases of using just `/usr/bin/foo ix,` but most cases we have are `/usr/bin/foo ixr,` like this so I think it's fine, but we should try to be consistent everywhere.

Reading the apparmor manpage though it's unclear that `ix` implies `r`, AFAICT it seems to me that either:
* `ix` implies `r` but this is undocumented
* just using `ix` only allows calling execve on the path, not `read()` on that path, which IMHO is kinda a pointless distinction, I'm not sure why we would only want to allow executing a program and not reading that binary file? Also it feels like executing the program enables reading it anyways too since symbols have to be loaded etc so this could be either not enforced or easily bypassed in practice?

**Review by anonymouse64 — APPROVED** [review]
lgtm to allow using blkid with block-devices interface

File: interfaces/builtin/block_devices.go
Comment (setharnold) [reply interfaces/builtin/block_devices.go:104]: `r` has a complicated history. Once upon a time I believe it controlled the dumpable flag on the process, though that doesn't seem to work in a test on 5.4.0-77-generic today. I also think that kernel binfmt elf changes have added and removed the 'read' requirement when executing processes, so it might not be needed in some environments but needed in others.

Sometimes secrets are embedded directly in executables, and the executable readability is used to hide the secret. This is probably not a great design pattern but I understand why folks use it.

I wish I'd kept better notes along the way. Anyway, I tend to use all of `rmix` when I'm writing profiles for myself.

Comment (anonymouse64) [reply interfaces/builtin/block_devices.go:104]: @setharnold thanks for the insight; so what I think we should do in this case is use rmix or rix for all binaries like this, but do so in a different PR so that the security team can audit the set of programs that we are allowing with `m` since that flag seems possibly somewhat dangerous for some problematic binaries?

In any case, I don't think that's a reason to hold up this PR which we want to include in an imminent 2.53.2 release, and we have plenty of other programs using `rix`, or more commonly `ixr`, I don't think `blkid` should be subject to additional consideration here regarding `rix` vs `rmix`

**Review by metorino — APPROVED** [review]
LGTM, thanks!

**Comment by mvo5** [comment]
I removed "needs security review" because Emilia reviewed it.

**Pull request merged**
